OlympGuide.Application/Features/Reservation/ReservationService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using OlympGuide.Domain.Features.Reservation;
+using OlympGuide.Domain.Features.User;
+
+namespace OlympGuide.Application.Features.Reservation
+{
+    public class ReservationService(IReservationRepository repository, IUserService userService) : IReservationService
+    {
+        private readonly IReservationRepository _repository = repository;
+        public async Task<List<ReservationType>> GetAllReservations()
+        {
+            var user = await userService.GetCurrentUserFromUserContext();
+            if(user.Roles.Contains(UserRole.Administrator))
+            {
+                return await _repository.GetAllReservations();
+            }
+            return await _repository.GetReservationsByUser(user);
+        }
+
+        public async Task<ReservationType> GetReservationById(Guid reservationId)
+        {
+            if (reservationId == Guid.Empty)
+            {
+                throw new ArgumentException("Guid must no be null");
+            }
+
+            try
+            {
+                return await _repository.GetReservationById(reservationId);
+            }
+            catch (InvalidOperationException)
+            {
+                throw new NoReservationFoundException(reservationId);
+            }
+        }
+
+        public async Task<List<ReservationType>> GetReservationsBySportField(Guid sportFieldId)
+        {
+            if (sportFieldId == Guid.Empty)
+            {
+                throw new ArgumentException("Guid must no be null");
+            }
+
+            try
+            {
+                return await _repository.GetReservationsBySportField(sportFieldId);
+            }
+            catch (InvalidOperationException)
+            {
+                throw new InvalidOperationException(String.Format("Reservation for SportField with id: {0} was not found", sportFieldId.ToString()));
+            }
+        }
+
+        public async Task<List<ReservationType>> GetReservationsByUser(Guid userId)
+        {
+            if (userId == Guid.Empty)
+            {
+                throw new ArgumentException("Guid must no be null");
+            }
+            var user = await userService.GetUserProfile(userId);
```

Comment:
will throw an UserNotFoundException if the user can not be found. => try{}catch{}